src/routes/settings/container/SettingsContainer.js

```diff
@@ -0,0 +1,25 @@
+// @flow
+
+import Settings from '../components/Settings'
+import { withTheme } from 'styled-components/native'
+import type { NavigationScreenProp } from 'react-navigation'
+import { translate } from 'react-i18next'
+import { connect } from 'react-redux'
+import type { StateType } from '../../../modules/app/StateType'
+
+type OwnPropsType = {| navigation: NavigationScreenProp<*> |}
+
+type PropsType = {
+  navigation: NavigationScreenProp<*>
+}
+
+const mapStateToProps = (state: StateType) => {
+  return {
+    language: state.cityContent.language
+  }
+}
+
+const ThemedSettings = withTheme<PropsType, _>(translate('settings')(Settings))
+export default connect<PropsType, OwnPropsType, _, _, _, _>(mapStateToProps)(
```

Comment:
Haha, this is super ugly :D but oh well^^

Reply:
Yes it is.. ... complicated